lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -248,7 +248,17 @@ def get_or_create_cached_value():
                         return_value = func(*args, **kwargs)
 
                 messages = cached_func.message_call_stack._most_recent_messages
-                cache.write_result(value_key, return_value, messages)
+                try:
+                    cache.write_result(value_key, return_value, messages)
+                except TypeError as err:
+                    module_name = (
+                        f"{return_value.__module__}.{type(return_value).__name__}"
+                    )
+                    if module_name == "snowflake.snowpark.session.Session":
```

Comment:
Can you use `type_util.is_type` function?

Reply:
@sfc-gh-kbregula thank you for the review 👍 I've used `type_util.is_type` here 👍